### ISSUE TITLE
0.3.0 Save prompt name - can't save image.

### ISSUE THREAD

**Issue by Zimbakin** [divamgupta/diffusionbee-stable-diffusion-ui#118]
0.3.0 - Save with prompt name: unable to save.
Most images can't be saved possibly due to character count limit being exceeded.

Have to manually rename files now to save.

Maybe use the seed number as the file name with versioning, or a combination of first 20 charters + seed?

**Comment by divamgupta** [comment]
What error do you see?

**Comment by Zimbakin** [comment]
The save button is ghosted out.  M1 OS Monterey 12.6

> eg. a photorealistic dramatic hyperrealistic render of the mandalorian bounty hunter, ultra realistic details, well worn, rust, oil stains by wlop, greg rutkowski, alphonse mucha vitaly bulgarov and mike nash, beautiful dramatic dark moody tones and lighting, cinematic atmosphere, studio lighting, global illumination, shadows, dark background

This is 340 characters.
If I reduce the file name to 251 characters, the save button turns blue and I can now save.

**Comment by Zimbakin** [comment]
... Seems the character count is off by 3 characters.
For any prompt greater than 255 characters, you need to delete x3 characters in order to save.

**Comment by Leland** [comment]
Yes, the file length limit for both HFS+ and APFS is 255 characters. In practice, it's a few shorter.

The quick solution to this issue is to truncate the name to 251 characters.
